zio-schema-derivation/shared/src/main/scala-2/zio/schema/DeriveSchema.scala

```diff
@@ -65,11 +68,13 @@ object DeriveSchema {
       } else if (isCaseClass(tpe)) deriveRecord(tpe, stack)
       else if (isSealedTrait(tpe))
         deriveEnum(tpe, stack)
+      else if (isSealedAbstractClass(tpe))
```

Comment:
Is there a need to make a difference here? How about just `isSealed` and join the two cases

Reply:
No it turned out to be the same. I think keeping the two explicit `isXXX` functions is good for understandability, but we can merge the two branches